functions/supplier/zeromq-supplier/src/main/java/org/springframework/cloud/fn/supplier/zeromq/ZeroMqSupplierProperties.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2016-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.fn.supplier.zeromq;
+
+import java.time.Duration;
+
+import javax.validation.constraints.NotEmpty;
+
+import org.hibernate.validator.constraints.Range;
+import org.zeromq.SocketType;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.validation.annotation.Validated;
+
+/**
+ *
+ * @author Daniel Frey
+ * @since 3.1.0
+ */
+@ConfigurationProperties("zeromq.supplier")
+@Validated
+public class ZeroMqSupplierProperties {
+
+	/**
+	 * The {@link SocketType} the connection should make.
+	 */
+	private SocketType socketType = SocketType.SUB;
+
+	/**
+	 * Connection URL for to the ZeroMQ TCP Server.
+	 */
+	private String connectUrl;
+
+	/**
+	 * Bind Port for connecting to the ZeroMQ TCP Server.
```

Comment:
We probably need to mention that "0 means random".
Plus it looks like your description is misleading: we don't connect to the server - we really create a server with bind port.
In fact it doesn't look like in ZeroMQ socket API there is a concept like client/server: there is connecting socket and bound one. Who is the server or client is a target application decision.

(Sorry for being for nasty, but I foresee some questions from end-users which would be better to avoid if we are good with our docs)